crml/sylo/src/device.rs

```diff
@@ -36,26 +32,87 @@ decl_storage! {
 	}
 }
 
-decl_event!(
-	pub enum Event<T> where <T as frame_system::Trait>::Hash, <T as frame_system::Trait>::AccountId {
-		DeviceAdded(AccountId, Hash, u32),
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// There are no devices registered for user (missing user_id in Devices)
+		UserId,
+		/// Device is already registered to user (device_id is already in use)
+		DeviceId,
+		/// A user can't have more than MAX_DEVICES registered devices
+		MaxDevice,
 	}
-);
+}
 
 impl<T: Trait> Module<T> {
 	pub fn append_device(user_id: &T::AccountId, device_id: u32) -> DispatchResult {
 		let mut devices = <Devices<T>>::get(user_id);
 
-		ensure!(!devices.contains(&device_id), "Device Id already in use");
-		ensure!(
-			devices.len() <= MAX_DEVICES,
-			"User has registered up to the maximum number of devices"
-		);
+		ensure!(!devices.contains(&device_id), Error::<T>::DeviceId);
+		ensure!(devices.len() < MAX_DEVICES, Error::<T>::MaxDevice);
```

Comment:
The length check was incorrect before as it was possible to add `MAX_DEVICES + 1` devices

Reply:
Eagle eye! Nice catch!